Source/Dafny/AST/DafnyAst.cs

```diff
@@ -4312,9 +4312,12 @@ public TopLevelDecl(IToken tok, string name, ModuleDefinition enclosingModule, L
             return externArgs[0].AsStringLiteral() + "." + externArgs[1].AsStringLiteral();
           }
         }
-
-        return DafnyOptions.O.Compiler.GetCompileName(EnclosingModuleDefinition.IsDefaultModule,
+        if (DafnyOptions.O.Compiler != null) {
+          return DafnyOptions.O.Compiler.GetCompileName(EnclosingModuleDefinition.IsDefaultModule,
           EnclosingModuleDefinition.CompileName, CompileName);
+        } else {
+          return Declaration.IdProtect(EnclosingModuleDefinition.CompileName) + "." + Declaration.IdProtect(CompileName);
+        }
```

Comment:
Why do we not instantiate the compiler?

Reply:
I don't think the AST is the right place to instantiate a compiler anyway, it should be instantiated in the options if really instantiated manually. I we don't want the compiler to be null, I think we should always instantiate a default C# compiler. Would you prefer that?